keps/sig-api-machinery/0015-dry-run.md

```diff
@@ -110,9 +110,32 @@ type AdmissionRequest struct {
 }
 ```
 
+## Generated values
+
+Some values of the object are typically generated before the object is persisted:
+- generateName can be used to assign a unique random name to the object,
+- creationTimeStamp records the time of creation,
+- UID uniquely identifies the object and is randomly generated (non-deterministic),
+- resourceVersion tracks the persisted version of the object.
+
+Most of these values are not useful in the context of dry-run, and could create
+some confusion. The UID and the generated name would have a different value in a
+dry-run and non-dry-run creation. These values will be left empty when
+performing a dry-run.
+
+CreationTimeStamp is also generated on creation, but there are less ways to
+abuse it so it will be generated as it would for a regular request.
+
+ResourceVersion will also be left empty on creation.
```

Comment:
I agree that we should leave it empty on creation. And on updates, I think the value should be unchanged for dry-run requests

Reply:
@liggitt do you think this is the way we want to go? I think it would be better than leaking some information (probably incorrect information) about what the next rv is going to be

Reply:
yeah, leaving empty/unchanged makes sense to me